client/ayon_usd/standalone/usd/pinning/pinning_file_helper.py

```diff
@@ -0,0 +1,287 @@
+
+
+def remove_root_from_dependency_info(
+    dependency_info: Dict[str, str], root_info: Dict[str, str]
+) -> Dict[str, str]:
+    """Re-roots a path dict so that a Ayon_Usd_Resolver can use its own system for root replace to resolve a given path
+
+    Args:
+        dependency_info: dict generated by `get_asset_dependencies()`
+        root_info: a flat dict containing str: root identifier {root[work]}, str: associated root path /path/to/root
+
+    Returns:
+        The input dependency_info dict with replaced root_info to obtain root-less paths
+    """
+
+    if not root_info or not dependency_info:
+        return {}
+
+    replacements = {
+        re.escape(path_portion): replacer
+        for replacer, path_portion in root_info.items()
+    }
+
+    pattern = re.compile("|".join(f"({pat})" for pat in replacements))
+
+    def _replace_match(match: re.Match):
+        for _, group in enumerate(match.groups(), start=1):
+            if group:
+                replacment = "{root[" + replacements[re.escape(group)] + "]}"
+                return replacment
+        return match.group(0)
+
+    root_less_dict = {}
+
+    for key, path in dependency_info.items():
+        new_path = pattern.sub(_replace_match, path)
+        root_less_dict[key] = new_path
+
+    return root_less_dict
+
+
+def _get_prim_prop_data(prim: Sdf.PrimSpec, layer: Sdf.Layer) -> List[str]:
+    prop_data = []
+    for prop in prim.properties:
+        if not isinstance(prop, Sdf.AttributeSpec):
+            continue
+
+        if prop.typeName != Sdf.ValueTypeNames.Asset:
+            continue
+
+        default_val = prop.default
+        if default_val:
+            if hasattr(default_val, "__iter__"):
+                prop_data.extend(default_val)
+            else:
+                prop_data.append(default_val.path)
+        else:
+            time_samples = layer.ListTimeSamplesForPath(prop.path)
+            for time in time_samples:
+                value = layer.QueryTimeSample(prop.path, time)
+                prop_data.append(value.path)
+
+    return prop_data
+
+
+def _get_prim_spec_hierarchy_external_refs(
+    prim: Sdf.PrimSpec, layer: Sdf.Layer
+) -> List[str]:
+    file_list: List[str] = _get_prim_prop_data(prim, layer)
+
+    for child_prim in prim.nameChildren:
+        file_list.extend(_get_prim_spec_hierarchy_external_refs(child_prim, layer))
+    return file_list
+
+
+def _remove_sdf_args(ref: str) -> str:
+    uri = re.sub(re.compile(r":SDF_FORMAT_ARGS.*$"), "", ref)
+    return uri
+
+
+def _resolve_udim(udim_identifier: str, layer: Sdf.Layer) -> Dict[str, str]:
+    udim_data = {}
+    resolved_udims = UsdShade.UdimUtils.ResolveUdimTilePaths(
+        str(udim_identifier), layer
+    )
+
+    for resolved_udim_path in resolved_udims:
+        udim_replaced_identifier = udim_identifier.replace(
+            "<UDIM>",
+            resolved_udim_path[1],
+        )
+
+        udim_data[udim_replaced_identifier] = resolved_udim_path[0]
```

Comment:
```suggestion
    for path, tile in resolved_udims:
        udim_tile_identifier = udim_identifier.replace("<UDIM>", tile)
        udim_data[udim_tile_identifier] = path
```

Would this work and be more readable?

Reply:
Looks nice. 
Done in external commit.